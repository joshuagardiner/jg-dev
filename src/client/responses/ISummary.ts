/**
 * The response object the API returns.
 */
export interface ISummary {
  /**
   * The address.
   */
  address: string;

  /**
   * The email.
   */
  email: string;

  /**
   * The name.
   */
  name: string;

  /**
   * The role.
   */
  role: string;

  /**
   * The telephone.
   */
  telephone: string;

  /**
   * The text.
   */
  text: string[];
}
