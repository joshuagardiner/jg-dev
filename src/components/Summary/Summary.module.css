.avatar {
  border-radius: 50%;
  display: "inherit";
  margin-left: "auto";
  margin-right: "auto";
  position: "relative";
}

.contactInformation {
  font-size: 1rem;
}

.description {
  display: inline-block;
  font-size: 1.125rem;
  line-height: 2.75rem;
  position: relative;
  width: 80%;
}

.role {
  font-size: 1.25rem;
  margin-top: -1.75%;
  text-transform: uppercase;
}

.title {
  font-size: 3.5rem;
  margin-top: 7.5%;
}

@media screen and (max-width: 768px) {
  .description {
    width: 100%;
  }
}

@media screen and (max-width: 425px) {
  .contactInformation {
    font-size: 0.66rem;
  }

  .description {
    font-size: 0.75rem;
    line-height: 1.5rem;
    width: 100%;
  }

  .role {
    font-size: 1rem;
  }

  .title {
    font-size: 2rem;
  }
}
