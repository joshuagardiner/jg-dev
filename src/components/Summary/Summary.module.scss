.avatar {
  border-radius: 50%;
}

.avatarContainer {
  height: 21.875rem;
  margin-left: auto;
  margin-right: auto;
  margin-top: -10.938rem;
  width: 21.875rem;
}

.contactInformation {
  font-size: 1rem;
}

.description {
  display: inline-block;
  font-size: 1.125rem;
  line-height: 2.75rem;
  width: 80%;
}

.role {
  font-size: 1.25rem;
  margin-top: -1.75%;
  text-transform: uppercase;
}

.socialMediaIcons {
  flex-direction: row;
  line-height: 1.75em;
}

.socialMediaIcon,
.socialMediaIcon:hover {
  background-color: inherit;
  border: none;
  border-radius: 50%;
  color: #3c4858;
  cursor: pointer;
  font-size: 2.188rem;
}

.title {
  font-size: 3.5rem;
  margin-top: 5%;
}

@media screen and (max-width: 768px) {
  .avatarContainer {
    height: 15.625rem;
    margin-top: -7.813rem;
    width: 15.625rem;
  }

  .contactInformation {
    font-size: 0.83rem;
  }

  .description {
    width: 100%;
  }

  .socialMediaIcon,
  .socialMediaIcon:hover {
    font-size: 1.563rem;
    padding: 0.188rem;
  }
}

@media screen and (max-width: 425px) {
  .avatarContainer {
    height: 7.813rem;
    margin-top: -3.907rem;
    width: 7.813rem;
  }

  .contactInformation {
    font-size: 0.66rem;
  }

  .description {
    font-size: 0.75rem;
    line-height: 1.5rem;
  }

  .role {
    font-size: 1rem;
  }

  .title {
    font-size: 2rem;
  }
}
