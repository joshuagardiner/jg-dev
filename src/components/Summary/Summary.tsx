import React from "react";
import {
  Text,
  Title,
  Title as Role,
} from "@joshuagardiner/typescript-component-library";
import Image from "next/image";
import styling from "./Summary.module.css";

/**
 * The Summary component is responsible for rendering the Avatar component
 * and contact details / bio content that sits within the application's main Body
 * of content.
 *
 * @returns the Summary component.
 */
export const Summary: React.FC = () => {
  return (
    <>
      <div
        style={{
          display: "inherit",
          height: "17vw",
          marginLeft: "auto",
          marginRight: "auto",
          position: "relative",
          width: "17vw",
        }}
      >
        <Image
          className={styling.avatar}
          src={"/avatar.jpg"}
          alt={"jg_avatar"}
          layout="fill"
          objectFit="contain"
        />
      </div>
      <Title fontSize={"3.5vw"} marginTop={"7.5%"} text={"Joshua Gardiner"} />
      <Role
        fontSize={"1.25vw"}
        marginTop={"-1.75%"}
        text={"Software Development Engineer"}
        textTransform={"uppercase"}
      />
      <div className={styling.contactInformation}>
        <p>37 Oakwood Gardens, Halifax, HX28HB</p>
        <p>joshuagardiner@live.com</p>
        <p>+44 7947 633 294</p>
      </div>
      <Text
        fontSize={"1.125vw"}
        lineHeight={"2.5vw"}
        maxWidth={"75%"}
        text={`An experienced software development engineer with a passion for developing web applications. Well-versed in a variety of technologies and writing code to create applications that are reliable and user-friendly.`}
      />
      <Text
        fontSize={"1.125vw"}
        lineHeight={"2.5vw"}
        maxWidth={"75%"}
        text={`A budding leader with a proven ability to mentor and motivate colleagues, who can communicate confidently, problem-solve effectively and utilise initiative to deliver applications that are tailored to organizational needs using best practices and the best tool for the job.`}
      />
    </>
  );
};
