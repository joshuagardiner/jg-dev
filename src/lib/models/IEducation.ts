/**
 * The Education model.
 */
export interface IEducation {
  /**
   * The course studied.
   */
  courseOfStudy: string;

  /**
   * The grades achieved.
   */
  grades: string[];

  /**
   * The place of study.
   */
  placeOfStudy: string;

  /**
   * The tenure of study.
   */
  tenure: string;
}
