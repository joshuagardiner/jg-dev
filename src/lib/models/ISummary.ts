/**
 * The Summary model.
 */
export interface ISummary {
  /**
   * The address.
   */
  address: string;

  /**
   * The content.
   */
  content: ISummaryContent[];

  /**
   * The email.
   */
  email: string;

  /**
   * The name.
   */
  name: string;

  /**
   * The role.
   */
  role: string;

  /**
   * The telephone.
   */
  telephone: string;
}

/**
 * The ISummaryContent model.
 */
export interface ISummaryContent {
  /**
   * The identifier.
   */
  id: string;

  /**
   * The value.
   */
  value: string;
}
