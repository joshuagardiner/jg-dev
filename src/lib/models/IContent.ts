import { IExperience } from "./IExperience";
import { ISkills } from "./ISkills";
import { ISummary } from "./ISummary";

/**
 * The Content model.
 */
export interface IContent {
  /**
   * The experiences content.
   */
  experiences: IExperience[];

  /**
   * The skills content.
   */
  skills: ISkills;

  /**
   * The summary content.
   */
  summary: ISummary;
}
