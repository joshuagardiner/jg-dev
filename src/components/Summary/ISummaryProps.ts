import { ISummary } from "../../lib/models/ISummary";

/**
 * The properties available for the Summary component.
 */
export interface ISummaryProps {
  /**
   * The summary content.
   */
  summary: ISummary;
}
