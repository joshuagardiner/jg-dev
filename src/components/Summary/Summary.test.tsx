import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import { ISummary } from "../../lib/models/ISummary";
import { Summary } from "./Summary";

describe("Summary", () => {
  const OLD_OPEN = global.open;

  const summary: ISummary = {
    name: "Harry Potter",
    role: "Auror",
    address: "4 Privet Drive, Little Whinging, Surrey",
    email: "Harry.Potter@MoM.co.uk",
    telephone: "+44 1234 567 890",
    content: [
      { id: "1", value: "Test paragraph 1." },
      { id: "2", value: "Test paragraph 2." }
    ]
  };

  beforeEach(() => {
    global.open = jest.fn();
  });

  afterEach(() => {
    global.open = OLD_OPEN;
    jest.clearAllMocks();
  });

  it("renders the Summary component correctly.", () => {
    render(<Summary summary={summary} />);

    const summaryImage = screen.getByAltText("jg_avatar");
    const summaryTitle = screen.getByTestId("summary-title");
    const summaryRole = screen.getByTestId("summary-role");
    const summaryAddress = screen.getByTestId("summary-contact-address");
    const summaryEmail = screen.getByTestId("summary-contact-email");
    const summaryTelephone = screen.getByTestId("summary-contact-telephone");
    const summaryGitHubBtn = screen.getByTestId("summary-github-btn");
    const summaryLinkedInBtn = screen.getByTestId("summary-linkedin-btn");
    const summaryDescription = screen.getByTestId("summary-description");
    const summaryDescriptionContents = screen.getAllByTestId("summary-description-content");

    expect(summaryImage).toBeDefined();
    expect(summaryTitle.textContent).toEqual("Harry Potter");
    expect(summaryRole.textContent).toEqual("Auror");
    expect(summaryAddress.textContent).toEqual("4 Privet Drive, Little Whinging, Surrey");
    expect(summaryEmail.textContent).toEqual("Harry.Potter@MoM.co.uk");
    expect(summaryTelephone.textContent).toEqual("+44 1234 567 890");
    expect(summaryGitHubBtn).toBeDefined();
    expect(summaryLinkedInBtn).toBeDefined();
    expect(summaryDescription).toBeDefined();
    expect(summaryDescriptionContents[0].textContent).toEqual("Test paragraph 1.");
    expect(summaryDescriptionContents[1].textContent).toEqual("Test paragraph 2.");
  });

  it("should open a new tab when clicking the GitHub IconButton.", () => {
    render(<Summary summary={summary} />);

    const summaryGitHubBtn = screen.getByTestId("summary-github-btn");
    fireEvent.click(summaryGitHubBtn);

    expect(global.open).toHaveBeenCalledWith("https://github.com/joshuagardiner", "_blank");
  });

  it("should open a new tab when clicking the LinkedIn IconButton.", () => {
    render(<Summary summary={summary} />);

    const summaryLinkedInBtn = screen.getByTestId("summary-linkedin-btn");
    fireEvent.click(summaryLinkedInBtn);

    expect(global.open).toHaveBeenCalledWith(
      "https://www.linkedin.com/in/joshua-gardiner",
      "_blank"
    );
  });
});
