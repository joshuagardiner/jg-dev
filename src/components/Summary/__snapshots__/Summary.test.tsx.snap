// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary renders the Summary component unchanged. 1`] = `
<div>
  <div
    data-testid="summary-component"
  >
    <div
      style="display: inherit; height: 17vw; margin-left: auto; margin-right: auto; position: relative; width: 17vw;"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          alt="jg_avatar"
          class="avatar"
          data-nimg="fill"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
        />
        <noscript />
      </span>
    </div>
    <div
      class="tcl-title"
      data-testid="title"
    >
      <h1
        style="font-size: 3.5vw; margin-top: 7.5%;"
      >
        Joshua Gardiner
      </h1>
    </div>
    <div
      class="tcl-title"
      data-testid="title"
    >
      <h1
        style="font-size: 1.25vw; margin-top: -1.75%; text-transform: uppercase;"
      >
        Software Development Engineer
      </h1>
    </div>
    <div
      class="contactInformation"
      data-testid="summary-contact"
    >
      <p
        data-testid="summary-contact-address"
      >
        37 Oakwood Gardens, Halifax, HX28HB
      </p>
      <p
        data-testid="summary-contact-email"
      >
        joshuagardiner@live.com
      </p>
      <p
        data-testid="summary-contact-telephone"
      >
        +44 7947 633 294
      </p>
    </div>
    <div
      class="tcl-text"
      data-testid="text"
    >
      <p
        style="font-size: 1.125vw; line-height: 2.5vw; margin-left: auto; margin-right: auto; max-width: 75%;"
      >
        An experienced software development engineer with a passion for developing web applications. Well-versed in a variety of technologies and writing code to create applications that are reliable and user-friendly.
      </p>
    </div>
    <div
      class="tcl-text"
      data-testid="text"
    >
      <p
        style="font-size: 1.125vw; line-height: 2.5vw; margin-left: auto; margin-right: auto; max-width: 75%;"
      >
        A budding leader with a proven ability to mentor and motivate colleagues, who can communicate confidently, problem-solve effectively and utilise initiative to deliver applications that are tailored to organizational needs using best practices and the best tool for the job.
      </p>
    </div>
  </div>
</div>
`;
